bnding_box_yolo_format.py

```diff
@@ -0,0 +1,110 @@
+import os, json, requests
+from PIL import Image
+from io import BytesIO
+import pandas as pd
+import numpy as np
+from sys import argv
+
+def get_box_centers_all_emotions(sample_row):
+    '''
+    input: A dictionary of the format {emotion1: [{geometry_box1: [list of 4 corner coordinates],
+                                                   geometry_box1: [list of 4 corner coordinates],
+                                                   ...}]
+                                       emotion2: [...],
+                                       ...
+                                       emotionN: [...]
+                                       }
+    output: A dictionary of the format {emotion1: [(center1 coordinates), (center1 coordinates), ...],
+                                        emotion2: [...],
+                                        ...
+                                        emotionN: [...]
+                                       }
+    '''
+
+    lab = json.loads(sample_row)
+    centers = {}
+    for key, val in lab.items():
+        lab_df = pd.DataFrame(val)
+        centers[key] = lab_df.iloc[:, 0].apply(lambda s: get_box_center(s)).tolist()
+    return centers
+
+def get_box_center(box_row):
+    '''
+    input: A list of 4 coordinates, representing the four corners of the bounding box.
+        Each coordinate is a dictionary of {'x': xcoord, 'y': ycoord}
+    output: A tuple representing the coordinates of the center of the bounding box, and its width and height.
+    '''
+    x_coords = []
+    y_coords = []
+    for item in box_row:
+        x_coords.append(item['x'])
+        y_coords.append(item['y'])
+    x_center = (np.max(x_coords) - np.min(x_coords)) / 2 + np.min(x_coords)
+    y_center = (np.max(y_coords) - np.min(y_coords)) / 2 + np.min(y_coords)
+    box_width = np.max(x_coords) - np.min(x_coords)
+    box_height = np.max(y_coords) - np.min(y_coords)
+    return (x_center, y_center, box_width, box_height)
+
+def get_width_height(sample_row):
+    response = requests.get(sample_row)
+    if response.status_code == 200: # !!!
+        img = Image.open(BytesIO(response.content))
+    return (img.width, img.height)
+
+def get_yolo_formats(emotion_dict, total_width, total_height):
+    '''
+    Input: a dictionary of {emotion: [(box_x_center, box_y_center, box_width, box_height),
+                                      (box_x_center, box_y_center, box_width, box_height), ...]}
+    '''
+    yolo_formats_to_write = []
+    for emotion, boxes in emotion_dict.items():
+#         class_id = label_map['class_id'][label_map['emotion']==emotion].values[0]
+        if emotion == 'happy':
```

Comment:
A few suggestions to make this code easier to maintain: 

1) this whole script should just take whatever labelbox is giving to us, and convert it to 
emotion_as_a_string yolo_formatted coordinates. 

2) then we take the output from step 1, and we can convert the results into:
class_id yolo_formateed coordinates. 

This way, people know, oh, for the first step, we are just parsing out what labelbox is giving. In step 2, we then can convert any text label into any class_id by calling a function.